tests/accessor/accessor_default_values.h

```diff
@@ -0,0 +1,174 @@
+      REQUIRE(std::is_same_v<sycl::accessor<T>,
+                             sycl::accessor<T, detail::expected_dims>>);
+    }
+    section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for accessMode "
+                                    "template parameter. [generic accessor]");
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<
+              sycl::accessor<T, Dimension>,
+              sycl::accessor<T, Dimension, detail::expected_mode<T>>>);
+    }
+    section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for accessTarget "
+                                    "template parameter. [generic accessor]");
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<sycl::accessor<T, Dimension, AccessMode>,
+                             sycl::accessor<T, Dimension, AccessMode,
+                                            detail::expected_target>>);
+    }
+    section_name =
+        get_section_name<Dimension>(type_name, access_mode_name, target_name,
+                                    "Verify default value for isPlaceholder "
+                                    "template parameter. [generic accessor]");
+    SECTION(section_name) {
+      REQUIRE(std::is_same_v<sycl::accessor<T, Dimension, AccessMode, Target>,
+                             sycl::accessor<T, Dimension, AccessMode, Target,
+                                            detail::expected_placeholder>>);
+    }
+  }
+};
+
+/**
+ * @brief Struct with functor that will be used in type coverage to call all
+ *        verification functions.
+ * @tparam T Current data type
+ * @param type_name Current data type string representation
+ */
+template <typename T>
+class run_tests {
+ public:
+  void operator()(const std::string& type_name) {
+    // Type packs instances have to be const, otherwise for_all_combination
+    // will not compile
+    const auto access_modes = get_access_modes();
+    const auto dimensions = get_all_dimensions();
+    const auto targets = get_targets();
+
+    // Run test with non const data type
+    for_all_combinations<test_for_generic_acc, T>(access_modes, dimensions,
+                                                  targets, type_name);
+    for_all_combinations<test_for_host_acc, T>(access_modes, dimensions,
+                                               targets, type_name);
+    test_for_local_acc<T>(type_name);
+    // Run test with const data type
+    const auto const_type_name = "const " + type_name;
+    for_all_combinations<test_for_generic_acc, const T>(
+        access_modes, dimensions, targets, const_type_name);
+    for_all_combinations<test_for_host_acc, const T>(access_modes, dimensions,
+                                                     targets, const_type_name);
+    test_for_local_acc<const T>(const_type_name);
```

Comment:
Nit; what do you think about adding a blank line?

```suggestion
    // Run test with non const data type
    for_all_combinations<test_for_generic_acc, T>(access_modes, dimensions,
                                                  targets, type_name);
    for_all_combinations<test_for_host_acc, T>(access_modes, dimensions,
                                               targets, type_name);
    test_for_local_acc<T>(type_name);

    // Run test with const data type
    const auto const_type_name = "const " + type_name;
    for_all_combinations<test_for_generic_acc, const T>(
        access_modes, dimensions, targets, const_type_name);
    for_all_combinations<test_for_host_acc, const T>(access_modes, dimensions,
                                                     targets, const_type_name);
    test_for_local_acc<const T>(const_type_name);
```

Reply:
blank line has been added [here](https://github.com/KhronosGroup/SYCL-CTS/pull/335/commits/e2e50aa7cd5d3b4f749e1d1c3149f1c8a5d2c727#diff-9ed3eae8f2889bc04dfc0fad247a8fa4e381e1fff4144e0f1a23670d108f6fe9R163)